### PR TITLE
Support Python 3

### PR DESCRIPTION
**Pull request by hasegeli**

### CODE REVIEW

**Comment by hasegeli** [comment]
I made it depend on #13 to avoid rebase hassle.

File: igvm/commands.py [hunk 1]
```diff
@@ -420,3 +536,60 @@ def vm_rename(vm_hostname, new_hostname, offline=False):
...
+
+def _get_vm(hostname, ignore_reserved=False):
+    """Get a server from Serveradmin by hostname to return VM object
+
+    The function is accepting hostnames in any length as long as it resolves
+    to a single server on Serveradmin.
+    """
+    conditions = [ExactMatch(hostname)]
+    if hostname.endswith('.ig.local'):
+        conditions.append(ExactMatch(hostname[:-len('.ig.local')]))
```
Comment (kofrezo) [inline igvm/commands.py]: We should add a todo to remove this when we have all servers renamed properly

File: igvm/commands.py [hunk 2]
```diff
@@ -145,10 +158,113 @@ def vm_build(vm_hostname, localimage=None, nopuppet=False, postboot=None,
...
+    if not was_running:
+        offline = True
+
+    if not offline and newip:
+        raise IGVMError('Online migration cannot change IP address.')
+
+    if not offline and runpuppet:
+        raise IGVMError('Online migration cannot run Puppet.')
+
+    if not runpuppet and newip:
```
Comment (kofrezo) [inline igvm/commands.py]: Out of curiosity, whats the reason to run puppet but not change the IP on migration ?

File: igvm/migratevm.py [hunk 3]
```diff
@@ -2,126 +2,9 @@
 
 Copyright (c) 2018, InnoGames GmbH
 """
+# XXX: This module is for backwards compatibility.
```
Comment (kofrezo) [inline igvm/migratevm.py]: Where do we use this still ?

File: igvm/commands.py (hunk 1 above)
Comment (hasegeli) [reply igvm/commands.py]: This was actually to find the hosts without FQDN when the search is with FQDN.  We don't need this anymore.  So I am removing it, and putting the `TODO` to the place we append `.ig.local`.

File: igvm/commands.py (hunk 2 above)
Comment (hasegeli) [reply igvm/commands.py]: Maybe @TuxPowered42 knows?

File: igvm/migratevm.py (hunk 3 above)
Comment (hasegeli) [reply igvm/migratevm.py]: Any script that needs it.  This is the current place of the function.

File: igvm/commands.py (hunk 1 above)
Comment (hasegeli) [reply igvm/commands.py]: I prepared #18 for this.

File: igvm/commands.py (hunk 2 above)
Comment (TuxPowered42) [reply igvm/commands.py]: I don't know. It was always an option to run puppet after migration. I don't think there is any other use for it than to change IP address of a VM.

**Pull request merged**